RK-coeff-opt/oc_albrecht.m

```diff
@@ -24,12 +24,26 @@
 %=====================================================
 % Stage order conditions
 
-tau_2 = (c.^2)/factorial(2) - A*c;
-tau_3 = (c.^3)/factorial(3) - A*(c.^2)/factorial(2);
-tau_4 = (c.^4)/factorial(4) - A*(c.^3)/factorial(3);
-tau_5 = (c.^5)/factorial(5) - A*(c.^4)/factorial(4);
-tau_6 = (c.^6)/factorial(6) - A*(c.^5)/factorial(5);
-tau_7 = (c.^7)/factorial(7) - A*(c.^6)/factorial(6);
+c2_fac = (c .* c) ./ 2;
```

Comment:
Here are some profiling results I've obtained:

- For the form I've implemented, `oc_albrecht` takes 1.277 s. In detail:

  time | calls | line | code
  -- | -- | -- | --
  0.022 | 100001 | 27 | c2_fac = (c .* c) ./ 2;
  0.016 | 100001 | 28 | c3_fac = c .* c2_fac ./ 3;
  0.016 | 100001 | 29 | c4_fac = c .* c3_fac ./ 4;
  0.015 | 100001 | 30 | c5_fac = c .* c4_fac ./ 5;
  0.015 | 100001 | 31 | c6_fac = c .* c5_fac ./ 6;
  0.015 | 100001 | 32 | c7_fac = c .* c6_fac ./ 7;
    |   | 33 |  
  0.045 | 100001 | 34 | tau_2 = c2_fac - A * c;
  0.036 | 100001 | 35 | tau_3 = c3_fac - A * c2_fac;
  0.034 | 100001 | 36 | tau_4 = c4_fac - A * c3_fac;
  0.034 | 100001 | 37 | tau_5 = c5_fac - A * c4_fac;
  0.036 | 100001 | 38 | tau_6 = c6_fac - A * c5_fac;
  0.034 | 100001 | 39 | tau_7 = c7_fac - A * c6_fac;

- For the former form, `oc_albrecht` takes 7.130 s, 3.680 s are spent in `factorial`. In detail:

  time | Calls | line | code
  -- | -- | -- | --
  0.320 | 100001 | 41 | tau_2 = (c.^2)/factorial(2) - A*c;
  0.960 | 100001 | 42 | tau_3 = (c.^3)/factorial(3) - A*(c.^2)/factorial(2);
  1.227 | 100001 | 43 | tau_4 = (c.^4)/factorial(4) - A*(c.^3)/factorial(3);
  1.223 | 100001 | 44 | tau_5 = (c.^5)/factorial(5) - A*(c.^4)/factorial(4);
  1.228 | 100001 | 45 | tau_6 = (c.^6)/factorial(6) - A*(c.^5)/factorial(5);
  1.224 | 100001 | 46 | tau_7 = (c.^7)/factorial(7) - A*(c.^6)/factorial(6);

Running a full optimization, `factorial` was the function with the biggest self-time reported by the matlab profiler. That's why I wanted to remove that bottleneck - if I remember correctly, I got a speedup on the order of 15 seconds for a total runtime of 60 seconds.

Reply:
Moving the definitions of the `tau_i` down would be a bit annoying, since we would have to repeat them several times (depending on the stage order) in the current setup.